service-workers/service-worker/resources/request-end-to-end-worker.js

```diff
@@ -4,6 +4,8 @@ onmessage = function(e) {
   var message = e.data;
   if (typeof message === 'object' && 'port' in message) {
     port = message.port;
+
+    port.postMessage('received port');
   }
 };
```

Comment:
nit: technically this test is still flaky, as nothing is stopping a browser from killing the service worker as soon as the message event is handled, and restarting it before the fetch event is dispatch. To fix that you should call waitUntil on the message event with a promise you either don't resolve, or at least don't resolve until the fetch event handler.